File: objective-c/demo-auth-objc/DemoDataCell.m
//
//  DemoDataCell.m
//  Copyright © 2015-2016 Sequencing.com. All rights reserved
//

#import "DemoDataCell.h"

@implementation DemoDataCell

+ (CGFloat)heightForRow:(NSString *)text {
    UIFont *font = [UIFont systemFontOfSize:13.f];
    
    NSShadow *shadow = [[NSShadow alloc] init];
    shadow.shadowOffset = CGSizeMake(0, -1);
    shadow.shadowBlurRadius = 0.5;
    
    NSMutableParagraphStyle *paragraph = [[NSMutableParagraphStyle alloc] init];
    [paragraph setLineBreakMode:NSLineBreakByWordWrapping];
    [paragraph setAlignment:NSTextAlignmentLeft];
    
    NSDictionary *attributes = [NSDictionary dictionaryWithObjectsAndKeys:
                                font, NSFontAttributeName,
                                paragraph, NSParagraphStyleAttributeName,
                                shadow, NSShadowAttributeName, nil];
    
    CGRect rect = [text boundingRectWithSize:CGSizeMake(280, CGFLOAT_MAX)
                                     options:NSStringDrawingUsesLineFragmentOrigin | NSStringDrawingUsesFontLeading
                                  attributes:attributes
                                     context:nil];
    
    
    
    
    if (CGRectGetHeight(rect) < 42.960938f) {
        return 43.0f;
    } else {
        return CGRectGetHeight(rect) + 15;
    }
}



+ (NSString *)prepareText:(NSDictionary *)demoText {
    NSMutableString *preparedText = [[NSMutableString alloc] init];
    NSArray *keys = [NSArray arrayWithArray:[demoText allKeys]];
    NSArray *sortedKeys = [keys sortedArrayUsingSelector:@selector(localizedCaseInsensitiveCompare:)];
    
    for (NSString *key in sortedKeys) {
        id obj = [demoText objectForKey:key];
        NSString *tempString = [NSString stringWithFormat:@"%@: %@\n", key, obj];
        [preparedText appendString:tempString];
    }
    
    return preparedText;
}

+ (NSString *)prepareTextFromFile:(NSDictionary *)text AndFileType:(NSString *)fileType {
    NSMutableString *preparedText = [[NSMutableString alloc] init];
    
    if ([fileType containsString:@"Sample"]) {
        // working with sample file type
        NSString *friendlyDesk1 = [text objectForKey:@"FriendlyDesc1"];
        NSString *friendlyDesk2 = [text objectForKey:@"FriendlyDesc2"];
        NSString *tempString = [NSString stringWithFormat:@"%@\n%@", friendlyDesk1, friendlyDesk2];
        [preparedText appendString:tempString];
        
    } else {
        // working with own file type
        NSString *filename = [text objectForKey:@"Name"];
        NSString *tempString = [NSString stringWithFormat:@"%@", filename];
        [preparedText appendString:tempString];
        
    }
    
    return preparedText;
}

@end

File: swift/demo-auth-swift/DemoDataCell.swift
//
//  DemoDataCell.swift
//  Copyright © 2015-2016 Sequencing.com. All rights reserved.
//

import UIKit

class DemoDataCell: UITableViewCell {
    
    @IBOutlet weak var demoTextLabel: UILabel!
    
    
    class func heightForRow(text: String) -> CGFloat {
        let font = UIFont.systemFontOfSize(13)
        
        let shadow: NSShadow = NSShadow()
        shadow.shadowOffset = CGSizeMake(0, -1)
        shadow.shadowBlurRadius = 0.5
        
        let paragraph: NSMutableParagraphStyle = NSMutableParagraphStyle()
        paragraph.lineBreakMode = NSLineBreakMode.ByWordWrapping
        paragraph.alignment = NSTextAlignment.Left
        
        let options = unsafeBitCast(NSStringDrawingOptions.UsesLineFragmentOrigin.rawValue |
            NSStringDrawingOptions.UsesFontLeading.rawValue,
            NSStringDrawingOptions.self)
        
        let rect: CGRect = text.boundingRectWithSize(
            CGSizeMake(280, CGFloat.max),
            options: options, // NSStringDrawingOptions.UsesLineFragmentOrigin,
            attributes: [NSFontAttributeName: font, NSParagraphStyleAttributeName: paragraph, NSShadowAttributeName: shadow],
            context: nil)
        
        if CGRectGetHeight(rect) < 42.960938 {
            return 43.0
        } else {
            return CGRectGetHeight(rect) + 20
        }
    }
    
    
    class func prepareText(text: NSDictionary, FromFileType fileType: String) -> String {
        var preparedText = String()
        if fileType.containsString("Sample") {
            // working with sample file type
            let friendlyDesc1 = text.objectForKey("FriendlyDesc1") as? String
            let friendlyDesc2 = text.objectForKey("FriendlyDesc2") as? String
            preparedText += friendlyDesc1! + "\n" + friendlyDesc2!
        } else {
            // working with own file type
            let fileName = text.objectForKey("Name") as! String?
            preparedText += fileName!
        }
        return preparedText
    }
    
    
    class func prepareText(demoText: NSDictionary) -> String {
        var preparedText = String()
        let keys = NSArray(array: demoText.allKeys)
        let sortedKeys: NSArray = keys.sortedArrayUsingSelector("localizedCaseInsensitiveCompare:")
        for key in sortedKeys {
            let keyString = key as! String
            if demoText.objectForKey(key) is NSNull {
                preparedText = preparedText + keyString + ": \"" + "<null>\"" + "\n"
            } else if let value: String? = demoText.objectForKey(key) as! String? {
                preparedText = preparedText + keyString + ": \"" + value! + "\"\n"
            } else {
                preparedText = preparedText + keyString + ": \"" + "nil\"" + "\n"
            }
        }
        return preparedText
    }

}
